File: ゼミ用/05製造/eeldev/eelPy/api/request/DeleteAccountRequest.py
# /**
# * アカウント削除処理の値を保持する
# */
class DeleteAccountRequest:
    errorMsg = []

    # /**
    # * アカウント削除処理の値を格納します
    # */
    def __init__(self, userId):
        self.userId = userId

    def getUserId(self):
        return self.userId

    def getErrorMsg(self):
        return self.errorMsg

    # /**
    # * バリデーションチェックをします
    # */
    def validation(self):
        validationFlg = False

        try:
            # 入力項目チェック
            if (len(self.userId) == 0):
                self.errorMsg.append('エラー ：必須項目が入力されていません')
                validationFlg = True

            # ユーザIDの文字数チェック
            if (2000 < len(self.userId)):
                self.errorMsg.append('エラー ：登場人物説明は1200文字以内で入力してください')
                validationFlg = True

        except Exception as e:
            validationFlg = True

        return validationFlg

File: ゼミ用/05製造/api/src/request/DeleteAccountRequest.php
<?php
require_once dirname(__FILE__) . '/../common/Constant.php';

/**
 * アカウント削除処理の値を保持する
 */
class DeleteAccountRequest {
    private $userId;
    private $errorMsg = array();

    /**
     * アカウント削除処理の値を格納します
     */
    public function __construct($userId) {
        $this->userId = $userId;
    }

    public function getUserId() {
        return $this->userId;
    }

    public function getErrorMsg() {
        return $this->errorMsg;
    }

    /**
     * バリデーションチェックをします
     */
    public function validation() {
        $validationFlg = false;

        try {
            // 入力項目チェック
            if (empty($this->userId)) {
                array_push($this->errorMsg, 'エラー ：ユーザIDまたはパスワードが入力されていません');
                $validationFlg = true;
            }

            // ユーザIDの文字数チェック
            if (strlen($this->userId) != Constant::USER_ID_DIGIT + strlen(Constant::USER_ID_STR)) {
                array_push($this->errorMsg, 'エラー ：ユーザIDは8文字で入力してください');
                $validationFlg = true;
            }
        } catch (Exception $e) {
            $validationFlg = true;
        }

        return $validationFlg;
    }
}
